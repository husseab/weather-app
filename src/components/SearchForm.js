import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  return (
    <div className="search-form">
      <input
        className="search-input"
        type="text"
        onChange={handleInputChange}
        value={searchText}
      />
      <button className="search-button" type="submit" onClick={onSubmit}>
        Search
      </button>
    </div>
  );
}

export default SearchForm;

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};
