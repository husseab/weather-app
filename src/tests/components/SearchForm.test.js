import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "London",
    setSearchText: jest.fn(),
    handleCitySearch: jest.fn(),
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it(" the button displays the correct text.", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
      />
    );
    const button = screen.getByText(/Search/i);
    expect(button).toBeInTheDocument();
  });
});
