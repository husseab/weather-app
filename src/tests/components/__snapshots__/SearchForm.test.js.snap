// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      class="search-input"
      type="text"
      value="London"
    />
    <button
      class="search-button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
